library/core/src/fmt/mod.rs

```diff
@@ -337,10 +356,7 @@ impl<'a> Arguments<'a> {
     #[inline]
     #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
     #[rustc_const_unstable(feature = "const_fmt_arguments_new", issue = "none")]
-    pub const unsafe fn new_v1(
-        pieces: &'a [&'static str],
-        args: &'a [ArgumentV1<'a>],
-    ) -> Arguments<'a> {
+    pub const fn new_v1(pieces: &'a [&'static str], args: &'a [ArgumentV1<'a>]) -> Arguments<'a> {
```

Comment:
Was this previously already safe? I'm not sure I follow why we don't need the token as an arg here, too.

Reply:
I think all invariants are held with the panic case below. This was discussed and [decided against](https://github.com/rust-lang/rust/pull/83302#discussion_r606724772) in the previous PR, but I figured we can reconsider with the perf issue at hand.

Reply:
Just realized it should be okay to wrap the entire call with an unsafe block for this function. Want to try that and remove the panic?